resources/install.rb

```diff
@@ -79,7 +80,7 @@
     make_cmd << " USE_LINUX_TPROXY=#{use_linux_tproxy}"
     make_cmd << " USE_LINUX_SPLICE=#{use_linux_splice}"
 
-    extra_cmd = ' EXTRA=haproxy-systemd-wrapper' if node['init_package'] == 'systemd'
+    extra_cmd = ' EXTRA=haproxy-systemd-wrapper' if node['init_package'] == 'systemd' && !install_only
```

Comment:
Where ever we access a resource from the current/new resource we really need to access it via `new_resource.X` that  way Chef knows which resource to use. IIRC this should actually be an error on current Chef 13.